utils/utils.go

```diff
@@ -67,3 +71,48 @@ func Sha1File(path string) (string, error) {
 	io.Copy(hash, buff)
 	return hex.EncodeToString(hash.Sum(nil)), nil
 }
+
+// Extract hostnames from a list of pullhost args
+// ie. 'tcp://some.hostname.com:2375', extract 'some.hostname.com'
+func ParseHostnames(pullHosts []string) []string {
+	parsedHosts := make([]string, 0)
+
+	re := regexp.MustCompile("^tcp://(.+):2375$")
```

Comment:
Actually, to support everything, I'd just use the URL lib `https://golang.org/pkg/net/url/`